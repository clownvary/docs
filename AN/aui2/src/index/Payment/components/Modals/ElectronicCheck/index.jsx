import PropTypes from 'prop-types';
import React from 'react';
import { getPureNumber } from 'shared/utils/formatNumber';
import InputBankCard from 'react-base-ui/lib/components/InputBankCard';
import Modal from 'react-base-ui/lib/components/Modal';
import Checkbox from 'react-base-ui/lib/components/Checkbox';
import Button from 'react-base-ui/lib/components/Button';
import Dropdown from 'react-base-ui/lib/components/Dropdown';
import Alert from 'react-base-ui/lib/components/Alert';
import Popover from 'react-base-ui/lib/components/Popover';
import UIComponent from 'shared/components/UIComponent';
import AMIds from '../../../automationIds';

import './index.less';

const ACCOUNT_NUMBER = 'accountNumber';
const ROUTING_NUMBER = 'routingNumber';

const ACCOUNT_NUMBER_ERROR = 'Invalid Account Number';
const ROUTING_NUMBER_ERROR = 'Invalid Routing Number';
const defaultState = {
  [ACCOUNT_NUMBER]: '',
  [ROUTING_NUMBER]: ''
};

export default class ECheckModal extends UIComponent {

  static propTypes = {
    data: PropTypes.shape({
      showModel: PropTypes.bool.isRequired,
      accountTypeValue: PropTypes.string.isRequired,
      accountTypeList: PropTypes.shape({
        data: PropTypes.array.isRequired
      }).isRequired,
      saveInformation: PropTypes.bool.isRequired,
      newEcheckError: PropTypes.string.isRequired
    }).isRequired,

    title: PropTypes.string.isRequired,
    showPriorEcp: PropTypes.bool.isRequired,

    onCancel: PropTypes.func.isRequired,
    onSubmit: PropTypes.func.isRequired,
    onError: PropTypes.func.isRequired,
    changeSaveAccountInformation: PropTypes.func.isRequired,
    changeAccountType: PropTypes.func.isRequired
  }

  constructor(props) {
    super(props);
    this.state = {
      routingNumber: '',
      accountNumber: ''

    };
  }

  componentDidUpdate() {
    const { showModel } = this.props.data;

    if (!showModel) {
      this.clearError();
      this.resetData();
    }
  }

  submit = () => {
    const { accountNumber, routingNumber } = this.state;
    this.props.onSubmit(getPureNumber(accountNumber), getPureNumber(routingNumber));
  }

  changeEcheckAccountTypeHandler = (value) => {
    this.clearError();
    this.props.changeAccountType(value);
  }

  changeValue = (value, fieldName) => {
    const stateObject = {};
    const { newEcheckError } = this.props.data;
    stateObject[fieldName] = value;
    this.setState(stateObject);
    /* istanbul ignore if */
    if ((ROUTING_NUMBER === fieldName && newEcheckError === ROUTING_NUMBER_ERROR)
    || (ACCOUNT_NUMBER === fieldName && newEcheckError === ACCOUNT_NUMBER_ERROR)) {
      this.clearError();
    }
  }

  resetData = () => {
    this.setState({ ...defaultState });
  }

  closeNewECheck() {
    this.props.onCancel();
  }

  clearError = () => {
    if (this.props.data.newEcheckError) {
      this.props.onError('');
    }
  }

  render() {
    const {
      showModel,
      accountTypeValue,
      accountTypeList,
      saveInformation,
      newEcheckError
    } = this.props.data;
    const { accountNumber, routingNumber } = this.state;

    return (
      <div>
        <Modal
          title={this.props.title}
          shown={showModel}
          className="new-eCheck-modal"
          onClose={() => this.closeNewECheck()}
        >
          <div className="modal-body">
            {/* Error messages */}
            {
              /* eslint-disable react/jsx-boolean-value */
              newEcheckError ?
                <Alert type="danger" noClose>
                  <ul className="only-one-error">
                    <li>{newEcheckError}</li>
                  </ul>
                </Alert> : ''
              /* eslint-enable react/jsx-boolean-value */
            }
            <div className="form-group">

              {/* accountNumber row */}
              <div className={`form-row aaui-flexbox ${newEcheckError === ACCOUNT_NUMBER_ERROR ? 'form-error' : ''}`}>
                <label className="form-label-col" htmlFor={ACCOUNT_NUMBER}>
                  Account Number
                </label>
                <div className="form-entry-col">
                  {
                    showModel &&
                    <InputBankCard
                      id={ACCOUNT_NUMBER}
                      value={accountNumber}
                      maxLength={50}
                      data-qa-id={AMIds.modals.accountNumber}
                      onInput={(e, value) => { this.changeValue(value, ACCOUNT_NUMBER); }}
                    />
                  }
                </div>
                <div className="form-tooltip-col">
                  <span className="icon icon-info-circle popover-base" data-popover-trigger>
                    <Popover className="tooltip">
                      <div className="new_echeck_tip" />
                    </Popover>
                  </span>
                </div>
              </div>

              {/* routingNumber row */}
              <div className={`form-row  aaui-flexbox ${newEcheckError === ROUTING_NUMBER_ERROR ? 'form-error' : ''}`}>
                <label className="form-label-col" htmlFor={ROUTING_NUMBER}>
                  Routing Number
                </label>
                <div className="form-entry-col">
                  {showModel &&
                    <InputBankCard
                      id={ROUTING_NUMBER}
                      maxLength={10}
                      value={routingNumber}
                      data-qa-id={AMIds.modals.routingNumber}
                      onInput={(e, value) => { this.changeValue(value, ROUTING_NUMBER); }}
                    />
                  }
                </div>
                <div className="form-tooltip-col">
                  <span className="icon icon-info-circle popover-base" data-popover-trigger>
                    <Popover className="tooltip">
                      <div className="new_echeck_tip" />
                    </Popover>
                  </span>
                </div>
              </div>

              {/* echeck Type row */}
              {
                <div className="form-row aaui-flexbox">
                  <label className="form-label-col" htmlFor="accountType">
                    Account Type
                  </label>
                  <div className="form-entry-col">
                    <Dropdown
                      id="accountType"
                      value={accountTypeValue}
                      data-qa-id={AMIds.modals.accountType}
                      data={accountTypeList.data}
                      placeholder="Select Account type"
                      onChange={({ value }) => this.changeEcheckAccountTypeHandler(value)}
                      className="echeck-type-dropdown"
                    />
                  </div>
                </div>
              }

              {/* save echeck row */}
              <div className="form-row aaui-flexbox">
                <label className="form-label-col" htmlFor="newECheckCheckBox" />
                <div className="form-entry-col">
                  {
                    this.props.showPriorEcp ?
                      <Checkbox
                        id="newECheckCheckBox"
                        data-qa-id={AMIds.modals.saveInformation}
                        size="m"
                        checked={saveInformation}
                        onChange={e =>
                          this.props.changeSaveAccountInformation(e.target.checked)
                        }
                      >
                        Save account information
                    </Checkbox> : ''
                  }
                </div>
              </div>
            </div>
          </div>
          <div className="modal-footer">
            <Button onClick={() => this.closeNewECheck()}>Cancel</Button>
            <Button
              type="strong"
              onClick={this.submit}
              disabled={!accountNumber || !routingNumber || newEcheckError}
            >
              OK
            </Button>
          </div>
        </Modal>
      </div>
    );
  }
}
