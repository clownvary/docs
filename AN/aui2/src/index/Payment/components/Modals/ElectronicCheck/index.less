@import '~shared/style/variables/index.less';
/* stylelint-disable */

@image-url: '../../../../../shared/images';

.new-eCheck-modal {
  .echeck-type-dropdown {
    width: 180px;
  }

  .alert {
    margin-bottom: 30px;

    .only-one-error {
      display: inline-block;
      width: 520px;
      padding-left: 10px;
      margin: 0;
      line-height: 1.4;
      vertical-align: top;
      list-style: none;
    }
  }

  .form-group {
    .form-row {
      align-items: flex-start;
      min-height: 34px;
      margin: 10px 0 0;

      &:frist-child {
        margin: 0;
      }
    }

    .form-error {
      .checkbox__text:before,
      .input-group > input,
      .aaui-dropdown > button,
      .aaui-dropdown .aaui-dropdown__button-text {
        color: @state-danger-text;
        background-color: @state-danger-bg;
        border-color: @state-danger-border;
      }

      .checkbox {
        color: @state-danger-text;
      }

      .form-label-col {
        color: @state-danger-text;
      }
    }

    .form-label-col {
      flex: 0 0 141px;
      padding-right: 16px;
      margin-top: 2px;
      font-weight: @semibold;
      text-align: right;
    }

    .form-entry-col {
      flex: 0 0 320px;
      box-sizing: border-box;
    }

    .form-tooltip-col {
      flex: 0 0 36px;
      text-align: center;
    }

    .input-group {
      display: inline;
      width: 320px;
    }

    p {
      margin: 0;
      font-size: 14px;
      color: #db3811;
    }
  }

  .popover-base {
    margin-top: 6px;
    color: @color-blue;

    .popover--e {
      transform: translate(0, -20px);

      &:before,
      &:after {
        top: 20px;
      }
    }
  }

  .new_echeck_tip {
    width: 237px;
    height: 122px;
    background: url('@{image-url}/img_echeck_tips.png') top center no-repeat;
    background-size: 237px 122px;
  }
}
