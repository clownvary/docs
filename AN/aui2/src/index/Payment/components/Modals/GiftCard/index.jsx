import React from 'react';
import { connect } from 'react-redux';
import Input from 'react-base-ui/lib/components/Input';
import Modal from 'react-base-ui/lib/components/Modal';
import Checkbox from 'react-base-ui/lib/components/Checkbox';
import Button from 'react-base-ui/lib/components/Button';
import Dropdown from 'react-base-ui/lib/components/Dropdown';
import Alert from 'react-base-ui/lib/components/Alert';

import UIComponent from 'shared/components/UIComponent';

import {
  newGiftCardClose,
  newGiftCardIsOverrideMin,
  newGiftCardIsOverrideMax,
  changeGiftCardType,
  newGiftCardIsOverride,
  getNewGiftCardInfo,
  setNewGiftCardError
} from '../../../actions/paymentOptions/giftCard';

import './index.less';

export class GiftCard extends UIComponent {

  constructor(props) {
    super(props);
    this.state = {
      minNeedOverride: false,
      maxNeedOverride: false,
      giftCardNumber: '',
      giftCardErrorMsg: ''
    };
    this.refundBtnAble = true;
  }

  issueClick() {
    if (this.props.data.isNeedGiftCardNumber && !this.state.giftCardNumber) {
      this.validationNumber();
    } else {
      const {
        giftCardTypeListValue,
        isOverrideMinGiftCard,
        isOverrideMaxGiftCard } = this.props.data;
      const amount = +this.props.amount;
      const minGiftCardAmount = +this.getNewCardMaxMinAmount(false);
      const maxGiftCardAmount = +this.getNewCardMaxMinAmount();

      if (amount < minGiftCardAmount) {
        // istanbul ignore else
        if (!isOverrideMinGiftCard) {
          this.setState({ minNeedOverride: true });
          return false;
        }
      } else if (amount > maxGiftCardAmount) {
        // istanbul ignore else
        if (!isOverrideMaxGiftCard) {
          this.setState({ maxNeedOverride: true });
          return false;
        }
      }

      // istanbul ignore else
      if (this.refundBtnAble) {
        const params = {
          gc_number: this.state.giftCardNumber,
          gc_type_id: giftCardTypeListValue,
          gc_purchased_amount: this.props.amount
        };

        this.refundBtnAble = false;
        this.props.getNewGiftCardInfo({ params, amount: this.props.amount, key: this.props.index });
      }
    }

    return false;
  }

  validationNumber() {
    const errorMsg = this.state.giftCardNumber.length ? '' : 'Required';

    if (errorMsg) {
      this.setState({ giftCardErrorMsg: errorMsg });
    } else {
      this.setState({ giftCardErrorMsg: '' });
    }
  }

  changeGiftCardNumber(e, maxLength, isOnBlur = false) {
    const node = e.target;
    const oldLength = node.value.length;
    const oldIdx = node.selectionStart;
    let giftCardNumber = node.value;

    this.clearMaxMinError();

    if (isOnBlur) {
      this.validationNumber();
    }

    if (giftCardNumber.length > maxLength) {
      giftCardNumber = giftCardNumber.substr(0, maxLength);
    }

    this.setState({ giftCardNumber }, () => {
      if (e.type && e.type.toLowerCase() !== 'blur') {
        const newIdx = Math.max(0, (node.value.length - oldLength) + oldIdx);
        node.selectionStart = node.selectionEnd = newIdx;
      }
    });
    this.clearMaxMinError();
  }

  clearMaxMinError() {
    const {
      isOverrideMinGiftCard,
      isOverrideMaxGiftCard,
      newGiftCardError
    } = this.props.data;

    this.setState({
      minNeedOverride: false,
      maxNeedOverride: false
    });
    this.refundBtnAble = true;

    // istanbul ignore else
    if (isOverrideMinGiftCard) {
      this.props.newGiftCardIsOverrideMin(false);
    }

    // istanbul ignore else
    if (isOverrideMaxGiftCard) {
      this.props.newGiftCardIsOverrideMax(false);
    }
    if (newGiftCardError) {
      this.props.setNewGiftCardError('');
    }
  }

  closeNewGiftCard() {
    this.props.onCancelIssueCard();
    this.props.newGiftCardClose();
    this.clearMaxMinError();
  }

  changeTypeOption(value) {
    this.setState({
      giftCardNumber: ''
    });
    this.clearMaxMinError();
    this.props.changeGiftCardType(value);
  }

  getNewCardMaxMinAmount(isMax = true) {
    const { giftCardTypeList, giftCardTypeListValue } = this.props.data;
    let giftCardAmount = 0;

    giftCardTypeList.data.forEach((giftCard) => {
      if (giftCard.value === giftCardTypeListValue) {
        if (isMax) {
          giftCardAmount = giftCard.max_card_balance;
        } else {
          giftCardAmount = giftCard.min_sale_amount;
        }
      }
    });
    return giftCardAmount > 0 ? giftCardAmount : 0;
  }

  render() {
    const { giftCardLabel,
      giftCardTypeList,
      giftCardTypeListValue,
      isUseNewGiftCard,
      newGiftCardError,
      isNeedGiftCardNumber,
      isOverrideMinGiftCard,
      isOverrideMaxGiftCard,
      minOverrideHasAccess,
      minOverrideExplanation,
      maxOverrideHasAccess,
      maxOverrideExplanation } = this.props.data;

    const newGiftCardMin = `$${this.getNewCardMaxMinAmount(false).toFixed(2)}`;
    const newGiftCardMax = `$${this.getNewCardMaxMinAmount(true).toFixed(2)}`;

    return (
      <Modal
        title={`Issue New ${giftCardLabel}`}
        shown={isUseNewGiftCard}
        className="new-giftCard-modal"
        onClose={() => this.closeNewGiftCard()}
      >
        <div className="modal-body">
          {
            !newGiftCardError && this.state.minNeedOverride ?
              <Alert type="error" noClose>
                <ul className="only-one-error">
                  <li>{minOverrideExplanation} {newGiftCardMin}</li>

                  {
                    minOverrideHasAccess ?
                      <li>
                        <Checkbox
                          className="new-override"
                          checked={isOverrideMinGiftCard}
                          onChange={
                            () => this.props.newGiftCardIsOverrideMin(!isOverrideMinGiftCard)
                          }
                        >Override</Checkbox></li> : ''
                  }

                </ul>
              </Alert> : ''
          }

          {
            !newGiftCardError && this.state.maxNeedOverride ?
              <Alert type="error" noClose>
                <ul className="only-one-error">
                  <li>{maxOverrideExplanation} {newGiftCardMax}</li>

                  {
                    maxOverrideHasAccess ?
                      <li>
                        <Checkbox
                          className="new-override"
                          checked={isOverrideMaxGiftCard}
                          onChange={
                            () => this.props.newGiftCardIsOverrideMax(!isOverrideMaxGiftCard)
                          }
                        >Override</Checkbox></li> : ''
                  }

                </ul>
              </Alert> : ''
          }

          {
            newGiftCardError ?
              <Alert type="error" noClose>
                <ul className="only-one-error">
                  <li>{newGiftCardError}</li>
                </ul>
              </Alert> : ''
          }

          <div className="form-group">

            {/* giftCard Type row */}
            {
              <div className="form-row aaui-flexbox">
                <label className="form-label-col" htmlFor="giftCardType">
                  Card Type
                </label>
                <div className="form-entry-col">
                  <Dropdown
                    id="giftCardType"
                    value={giftCardTypeListValue}
                    data={giftCardTypeList.data}
                    placeholder="Select Account type"
                    onChange={({ value }) => this.changeTypeOption(value)}
                  />
                </div>
              </div>
            }

            {/* Number row */}
            {
              isNeedGiftCardNumber ?
                <div className={`form-row aaui-flexbox ${this.state.giftCardErrorMsg ? 'form-error' : ''}`}>
                  <label className="form-label-col" htmlFor="giftCardNumber">
                    Number
                </label>

                  <div className="form-entry-col">
                    <Input
                      id="giftCardNumber"
                      type="text"
                      maxLength={48}
                      value={this.state.giftCardNumber}
                      onBlur={e => this.changeGiftCardNumber(e, 48, true)}
                      onChange={e => this.changeGiftCardNumber(e, 48)}
                    />

                    {
                      this.state.giftCardErrorMsg ?
                        <p>{this.state.giftCardErrorMsg}</p> : ''
                    }

                  </div>
                </div>
                : ''
            }
          </div>
        </div>
        <div className="modal-footer">
          <Button onClick={() => this.closeNewGiftCard()}>Cancel</Button>
          <Button type="strong" onClick={() => this.issueClick()}>Issue</Button>
          {
            /*
            <Button type="strong" onClick={this.issueClick} disabled={this.state.refundBtnDisable}
            >Issue</Button>
            */
          }
        </div>
      </Modal>
    );
  }
}

export default connect(
  null,
  {
    newGiftCardClose,
    newGiftCardIsOverrideMin,
    newGiftCardIsOverrideMax,
    changeGiftCardType,
    newGiftCardIsOverride,
    getNewGiftCardInfo,
    setNewGiftCardError
  }
)(GiftCard);
