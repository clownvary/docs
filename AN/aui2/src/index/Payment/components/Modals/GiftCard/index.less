@import '~shared/style/variables/index.less';
/* stylelint-disable */

.new-giftCard-modal {
  .alert {
    margin-bottom: 30px;

    .only-one-error {
      display: inline-block;
      width: 520px;
      padding-left: 10px;
      margin: 0;
      line-height: 1.4;
      vertical-align: top;
      list-style: none;
    }

    .new-override {
      display: block;
      padding-top: 10px;
      font-size: 16px;
      line-height: 1.5;
      color: #313741;
    }
  }

  .form-group {
    .form-row {
      align-items: flex-start;
      min-height: 34px;
      margin: 10px 0 0;
    }

    .form-error {
      .checkbox__text:before,
      .input-group > input,
      .aaui-dropdown > button,
      .aaui-dropdown .aaui-dropdown__button-text {
        color: #d93a20 !important;
        background-color: #ffefed !important;
        border-color: #ff6046 !important;
      }

      .checkbox {
        color: #d93a20 !important;
      }
    }

    .form-label-col {
      flex: 0 0 141px;
      padding-right: 16px;
      margin-top: 2px;
      font-weight: @semibold;
      text-align: right;
    }

    .form-entry-col {
      flex: 0 0 300px;
    }

    .input-group {
      width: 300px;
    }

    p {
      margin: 0;
      font-size: 14px;
      color: #db3811;
    }
  }
}
