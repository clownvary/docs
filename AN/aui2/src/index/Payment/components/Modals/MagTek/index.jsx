import React from 'react';
import { connect } from 'react-redux';
import UIComponent from 'shared/components/UIComponent';
import { webStartServiceInstallHelpUrl } from 'shared/utils/webStartServiceHelper';
import { WSIHelper } from 'shared/components/WSIHelper';
import MagtekIPAD from './MagtekIPAD';
import MagtekDynamag from './MagtekDynamag';

import * as magtekActions from '../../../actions/modals/Magtek';
import { fetchCardtypeAction } from '../../../actions/modals/NewCreditCard';
import { retrieveCommunicationIframe } from '../../../utils/magtekFun';

export class MagTek extends UIComponent {

  componentDidMount() {
    retrieveCommunicationIframe();
  }

  /**
   * Windows Service client helper.
   */
  renderWSIHelper = (isShow) => {
    const { initialData } = this.props;
    const webStartServicehelpURL = `${initialData.helpFileFolder}${webStartServiceInstallHelpUrl}`;

    return isShow && <WSIHelper url={webStartServicehelpURL} />;
  }

  render() {
    const { data, initialData: { ccMagesafeDeviceType } } = this.props;
    const cardTypeList = data.get('cardTypeList') ? data.get('cardTypeList').data : [];
    return (
      <div className={data.get('isShowModal') ? '' : 'u-hidden'}>
        {
          ccMagesafeDeviceType === 'MagtekDynamag' ?
            <MagtekDynamag
              cardType={'Credit Card'}
              cardTypeList={cardTypeList}
              payAccmount={data.get('totalFee')}
              setCardInfo={this.props.setCardInfo}
              hideModalAction={this.props.hideMagtekModalAction}
              payment={this.props.payment}
              generateWalletID={this.props.generateWalletID}
              setAMSAccountInfo={this.props.setAMSAccountInfo}
              showModal={data.get('isShowModal')}
              serverError={data.get('error')}
              setServerError={this.props.setServerError}
              renderWSIHelper={this.renderWSIHelper}
              isShowSaveCardInformation={data.get('isShowSaveCardInformation')}
            /> :
            <MagtekIPAD
              cardType={'Credit Card'}
              cardTypeList={cardTypeList}
              payAccmount={data.get('totalFee')}
              setCardInfo={this.props.setCardInfo}
              hideModalAction={this.props.hideMagtekModalAction}
              payment={this.props.payment}
              generateWalletID={this.props.generateWalletID}
              setAMSAccountInfo={this.props.setAMSAccountInfo}
              showModal={data.get('isShowModal')}
              serverError={data.get('error')}
              setServerError={this.props.setServerError}
              renderWSIHelper={this.renderWSIHelper}
              isShowSaveCardInformation={data.get('isShowSaveCardInformation')}
            />
        }
      </div>
    );
  }

  componentWillMount() {
    this.props.fetchCardtypeAction();
  }
}

export default connect(
  null,
  {
    fetchCardtypeAction,
    ...magtekActions
  }
)(MagTek);
