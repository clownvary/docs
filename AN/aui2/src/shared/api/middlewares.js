import PromiseMiddleware from './promiseMiddleware';
import RouteMiddleware from './routeMiddleware';
import ThunkMiddleware from './thunkMiddleware';
import API from './API';

export default [
  new ThunkMiddleware(),
  new RouteMiddleware(),
  new PromiseMiddleware(new API())
];
