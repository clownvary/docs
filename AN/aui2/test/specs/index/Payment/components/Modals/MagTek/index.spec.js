import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import { fromJS } from 'immutable';
import configureStore from 'redux-mock-store';
import middlewares from 'shared/api/middlewares';
import { MagTek } from 'index/Payment/components/Modals/MagTek';

jest.mock('index/Payment/components/Modals/MagTek/MagtekIPAD', () => 'MagtekIPAD');
jest.mock('index/Payment/components/Modals/MagTek/MagtekDynamag', () => 'MagtekDynamag');

describe('index/Payment/components/Modals/MagTek', () => {

  let store;

  beforeEach(() => {
    store = configureStore(middlewares)({});
  });
  const actions = {
    fetchCardtypeAction: jest.fn(),
    setCardInfo: jest.fn(),
    hideMagtekModalAction: jest.fn(),
    payment: jest.fn(),
    generateWalletID: jest.fn(),
    setServerError: jest.fn()
  }
  const initialData = {
    ccMagesafeDeviceType: 'MagtekIPAD'
  };
  const props = {
    data: fromJS({
      isShowModal: false,
      totalFee: 111,
      error: ''
    }),
    initialData
  };
  const setup = (store, props) => mount(<MagTek store={store} {...props} {...actions} />);

  it('component should render correctly when ccMagesafeDeviceType is not MagtekDaynamg', () => {
    const component = setup(store, props);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should render correctly when ccMagesafeDeviceType is MagtekDaynamg', () => {
    const newProps = {
      data: props.data
        .set('cardTypeList', {data: []})
        .set('isShowModal', true),
      initialData: {
        ...initialData,
        ccMagesafeDeviceType: 'MagtekDynamag'
      }
    };
    const component = setup(store, newProps);

    expect(toJson(component)).toMatchSnapshot();
  });
});
