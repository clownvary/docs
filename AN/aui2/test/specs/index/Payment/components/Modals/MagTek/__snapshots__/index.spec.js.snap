// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index/Payment/components/Modals/MagTek component should render correctly when ccMagesafeDeviceType is MagtekDaynamg 1`] = `
<MagTek
  data={
    Immutable.Map {
      isShowModal: true,
      totalFee: 111,
      error: "",
      cardTypeList: Object {
        "data": Array [],
      },
    }
  }
  fetchCardtypeAction={[Function]}
  generateWalletID={[Function]}
  hideMagtekModalAction={[Function]}
  initialData={
    Object {
      "ccMagesafeDeviceType": "MagtekDynamag",
    }
  }
  payment={[Function]}
  setCardInfo={[Function]}
  setServerError={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className=""
  >
    <MagtekDynamag
      cardType="Credit Card"
      cardTypeList={Array []}
      generateWalletID={[Function]}
      hideModalAction={[Function]}
      payAccmount={111}
      payment={[Function]}
      renderWSIHelper={[Function]}
      serverError=""
      setCardInfo={[Function]}
      setServerError={[Function]}
      showModal={true}
    />
  </div>
</MagTek>
`;

exports[`index/Payment/components/Modals/MagTek component should render correctly when ccMagesafeDeviceType is not MagtekDaynamg 1`] = `
<MagTek
  data={
    Immutable.Map {
      isShowModal: false,
      totalFee: 111,
      error: "",
    }
  }
  fetchCardtypeAction={[Function]}
  generateWalletID={[Function]}
  hideMagtekModalAction={[Function]}
  initialData={
    Object {
      "ccMagesafeDeviceType": "MagtekIPAD",
    }
  }
  payment={[Function]}
  setCardInfo={[Function]}
  setServerError={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="u-hidden"
  >
    <MagtekIPAD
      cardType="Credit Card"
      cardTypeList={Array []}
      generateWalletID={[Function]}
      hideModalAction={[Function]}
      payAccmount={111}
      payment={[Function]}
      renderWSIHelper={[Function]}
      serverError=""
      setCardInfo={[Function]}
      setServerError={[Function]}
      showModal={false}
    />
  </div>
</MagTek>
`;
